nautobot_ssot_aristacv/diffsync/models/nautobot.py

```diff
@@ -40,27 +39,35 @@ class NautobotDevice(Device):
     @classmethod
     def create(cls, diffsync, ids, attrs):
         """Create device object in Nautobot."""
-        default_site_object = nautobot.verify_site(PLUGIN_SETTINGS.get("from_cloudvision_default_site", DEFAULT_SITE))
+        PLUGIN_SETTINGS = settings.PLUGINS_CONFIG["nautobot_ssot_aristacv"]
+        site_code, role_code = nautobot.parse_hostname(ids["name"].lower())
+        site_map = PLUGIN_SETTINGS.get("site_mappings")
+        role_map = PLUGIN_SETTINGS.get("role_mappings")
 
-        device_type_cv = attrs["device_model"]
+        if site_code and site_code in site_map:
+            site = nautobot.verify_site(site_map[site_code])
+        else:
+            site = nautobot.verify_site(PLUGIN_SETTINGS.get("from_cloudvision_default_site", DEFAULT_SITE))
 
-        device_type_object = nautobot.verify_device_type_object(device_type_cv)
-        device_role_object = nautobot.verify_device_role_object(
-            PLUGIN_SETTINGS.get("from_cloudvision_default_device_role", DEFAULT_DEVICE_ROLE),
-            PLUGIN_SETTINGS.get("from_cloudvision_default_device_role_color", DEFAULT_DEVICE_ROLE_COLOR),
-        )
+        if role_code and role_code in role_map:
+            role = nautobot.verify_device_role_object(
+                role_map[role_code],
+                PLUGIN_SETTINGS.get("from_cloudvision_default_device_role_color", DEFAULT_DEVICE_ROLE_COLOR),
+            )
+        else:
+            role = nautobot.verify_device_role_object(
+                PLUGIN_SETTINGS.get("from_cloudvision_default_device_role", DEFAULT_DEVICE_ROLE),
+                PLUGIN_SETTINGS.get("from_cloudvision_default_device_role_color", DEFAULT_DEVICE_ROLE_COLOR),
+            )
 
-        device_status = nautobot.verify_device_status(
-            PLUGIN_SETTINGS.get("from_cloudvision_default_device_status", DEFAULT_DEVICE_STATUS),
-            PLUGIN_SETTINGS.get("from_cloudvision_default_device_status_color", DEFAULT_DEVICE_STATUS_COLOR),
-        )
+        device_type_object = nautobot.verify_device_type_object(attrs["device_model"])
 
         new_device = OrmDevice(
-            status=device_status,
+            status=OrmStatus.objects.get(slug=attrs["status"]),
```

Comment:
Will the status attribute always be something that exists in Nautobot? I'm not sure what Status names CVP uses.

Reply:
The adapter specifically loads the status attribute on a Device as either `active` or `offline` depending upon the streaming status. If it's seen as active streaming it should be marked as active otherwise it's offline.

Reply:
Totally didn't know `offline` was a Status in Nautobot.